autoendpoint/src/extractors/subscription.rs

```diff
@@ -93,7 +93,29 @@ impl FromRequest for Subscription {
                 .map(|vapid| extract_public_key(vapid, &token_info))
                 .transpose()?;
 
-            trace!("Vapid: {:?}", &vapid);
+            trace!("raw vapid: {:?}", &vapid);
+
+            // Capturing the vapid sub right now will cause too much cardinality. Instead,
+            // let's just capture if we have a valid VAPID, as well as what sort of bad sub
+            // values we get.
+            if let Some(header) = vapid.clone() {
+                let sub = header
+                    .vapid
+                    .sub()
+                    .map_err(|e| {
+                        // Capture the type of error and add it to metrics.
+                        let mut tags = Tags::default();
+                        tags.tags
+                            .insert("error".to_owned(), e.as_metric().to_owned());
+                        metrics
+                            .clone()
+                            .incr_with_tags("notification.auth.error", Some(tags));
+                    })
+                    .unwrap_or_default();
+                // For now, record that we had a good (?) VAPID sub,
+                metrics.clone().incr("notification.auth.ok");
+                info!("VAPID sub: {:?}", sub)
```

Comment:
A few things here:

- The lack of `sub` isn't really [an actual error](https://datatracker.ietf.org/doc/html/rfc8292#section-2.1), maybe call this metric something else? Could it be a tag on the existing `notification.auth` metric (moving all this into `parse_vapid`)?
- If it doesn't make sense as part of `parse_vapid` I think ideally this extra work for metrics shouldn't occur until `validate_vapid_jwt` succeeds in validating
- Is the new `notification.auth.ok` totally necessary? We emit `updates.vapid.draft{version}` for successfully validated vapid, we could get a similar number by subtracting the no sub metric from those 2 combined